### Case study-3
Description about data: The data set (cust_seg) is information of 200
customers who were part of the last campaign conducted by one of the major
Bank-credit card division. The CMO would like to test below hypothesis based
on the data.


In [105]:
import os
import pandas as pd
import numpy as np
from scipy import stats

In [106]:
os.listdir()

['.ipynb_checkpoints',
 'Class exercise.ipynb',
 'Class1_BasicPython.ipynb',
 'cust_seg.csv',
 'DataTypes.ipynb',
 'EDA.ipynb',
 'EDAII.ipynb',
 'Hypothesis Testing- Case Study.pdf',
 'Hypothesis-Testing.ipynb',
 'Packages.ipynb',
 'Untitled.ipynb',
 'Viz.ipynb']

In [107]:
cust=pd.read_csv("cust_seg.csv")

In [108]:
cust.head()

custid  sex  AqChannel  region  Marital_status  segment  pre_usage  \
0      70    0          4       1               1        1         57   
1     121    1          4       2               1        3         68   
2      86    0          4       3               1        1         44   
3     141    0          4       3               1        3         63   
4     172    0          4       2               1        2         47   

   Post_usage_1month  Latest_mon_usage  post_usage_2ndmonth  
0                 52              49.2                 57.2  
1                 59              63.6                 64.9  
2                 33              64.8                 36.3  
3                 44              56.4                 48.4  
4                 52              68.4                 57.2

In [109]:
cust.tail()

custid  sex  AqChannel  region  Marital_status  segment  pre_usage  \
195      31    1          2       2               2        1         55   
196     145    1          4       2               1        3         42   
197     187    1          4       2               2        1         57   
198     118    1          4       2               1        1         55   
199     137    1          4       3               1        2         63   

     Post_usage_1month  Latest_mon_usage  post_usage_2ndmonth  
195                 59              62.4                 64.9  
196                 46              45.6                 50.6  
197                 41              68.4                 45.1  
198                 62              69.6                 68.2  
199                 65              78.0                 71.5

## Data Dictionary
 * custid: Customer ID/unique
 * sex:  Gender of customer /0:M, 1:F 
 * AqChannel: platform or channel they acqured the customer/1:office, 2:online, 3:DSA, 4:other channel 
 * region:  Demographic location/1: east, 2:west, 3:north, 4:south
 * Marital_status:  Marital status/1:Unmarried, 2:Married
 * segment:  1: Rich class, 2:middle class, 3:lower class
 * pre_usage: spending on credit card before campaign 
 * Post_usage_1month: spending on credit card after 1 month of the campaign launch
 * Latest_mon_usage: spending on credit card latest month after the campaign launch
 * post_usage_2ndmonth: spending on credit card after 2 months of the campaign launch

In [110]:
cust.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 10 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   custid               200 non-null    int64  
 1   sex                  200 non-null    int64  
 2   AqChannel            200 non-null    int64  
 3   region               200 non-null    int64  
 4   Marital_status       200 non-null    int64  
 5   segment              200 non-null    int64  
 6   pre_usage            200 non-null    int64  
 7   Post_usage_1month    200 non-null    int64  
 8   Latest_mon_usage     200 non-null    float64
 9   post_usage_2ndmonth  200 non-null    float64
dtypes: float64(2), int64(8)
memory usage: 15.8 KB


In [111]:
cust.describe()

custid        sex   AqChannel      region  Marital_status  \
count  200.000000  200.00000  200.000000  200.000000      200.000000   
mean   100.500000    0.54500    3.430000    2.055000        1.160000   
std     57.879185    0.49922    1.039472    0.724291        0.367526   
min      1.000000    0.00000    1.000000    1.000000        1.000000   
25%     50.750000    0.00000    3.000000    2.000000        1.000000   
50%    100.500000    1.00000    4.000000    2.000000        1.000000   
75%    150.250000    1.00000    4.000000    3.000000        1.000000   
max    200.000000    1.00000    4.000000    3.000000        2.000000   

          segment   pre_usage  Post_usage_1month  Latest_mon_usage  \
count  200.000000  200.000000         200.000000        200.000000   
mean     2.025000   52.230000          52.775000         63.174000   
std      0.690477   10.252937           9.478586         11.242137   
min      1.000000   28.000000          31.000000         39.600000   
25%      2.000000   44.000000          45.750000         54.000000   
50%      2.000000   50.000000          54.000000         62.400000   
75%      2.250000   60.000000          60.000000         70.800000   
max      3.000000   76.000000          67.000000         90.000000   

       post_usage_2ndmonth  
count           200.000000  
mean             58.052500  
std              10.426445  
min              34.100000  
25%              50.325000  
50%              59.400000  
75%              66.000000  
max              73.700000

## Q1.
Card usage has been improved significantly from last year usage which is 50. (Hint:
Comparing card usage of post campaign of 1 month with last year hypothesized value
50)\
### NB:\
one sample t-test is a statistical procedure where you wanted to test that where your population mean is different than a constant value(fix number). For example, a school wants to test that average mean of GPA for grad students is 3.0. They will use one sample t-test and can get the result.

Two sample t-test is also a statistical procedure where you are interested in testing whether these two population has the same mean or different mean. In the same example if the school is interested in testing that average GPA for science major and arts major is the same. Then they would have used two-sample t-test.


In [112]:
# Ho: avg usage in this yr=50 (There's no sig change in cr card usage)
# Ha: avg usage in this yr!=50 (There's sig change in cr card usage)
# Test: t-test=> 1 sample test (since ther's one pop/hypo mean)
# Decision rule:
#                          Let CI=99% (i.e. alpha=0.01)
#                               if p-value> alpha, accept Ho
#                               if p-value< alpha, reject Ho and accept Ha

In [113]:
stats.ttest_1samp(a=cust["Post_usage_1month"], popmean=50) # a :Sample observation, popmean : Expected value in null hypothesis Ho.

Ttest_1sampResult(statistic=4.140324966963025, pvalue=5.1209194607163335e-05)

p-value=0.0000512< alpha=0.01 ==> fail to accept Ho, and we accept Ha.\
i.e. There's sig change in Cr card usage from last yr

In [114]:
cust["pre_usage"].mean()   # mean far from 50, Ho 

52.23

## Q2
The last campaign was successful in terms usage of credit card.\
## NB:
paired samples t-test is used to compare the means of two samples when each observation in one sample can be paired with an observation in the other sample.\
commonly used in two scenarios:

* A measurement is taken on a subject before and after some treatment – e.g. the max vertical jump of college basketball players is measured before and after participating in a training program.

* A measurement is taken under two different conditions – e.g. the response time of a patient is measured on two different drugs.

In [115]:
# Ho: avg card usage before campaign= avg card usage before campaign  (There's no d/ce in pre and poat campaign card usage==> Campaign was not succ)
# Ha: avg card usage before campaign<> avg card usage before campaign  (There's d/ce in pre and poat campaign card usage==> Campaign was succ)
# Test: t-test=> paired sample t-test (since we are comparing sample pre and post campaign)
# Decision rule:
#                          Let CI=99% (i.e. alpha=0.01)
#                               if p-value> alpha, accept Ho
#                               if p-value< alpha, reject Ho and accept Ha
# comparing  pre_usage wrt   Post_usage_1month,   Latest_mon_usage,  and then wrt  post_usage_2ndmonth   

In [116]:
# stats.ttest_rel(a,b) - is a test for the null hypothesis that two related or repeated samples have identical average
p1month=stats.ttest_rel(cust["pre_usage"], cust["Post_usage_1month"])
p2month=stats.ttest_rel(cust["pre_usage"], cust["post_usage_2ndmonth"])
pLatest=stats.ttest_rel(cust["pre_usage"], cust["Latest_mon_usage"])
print(p1month)
print(p2month)
print(pLatest)

Ttest_relResult(statistic=-0.8673065458794775, pvalue=0.3868186820914985)
Ttest_relResult(statistic=-8.866832246938742, pvalue=4.295733828012836e-16)
Ttest_relResult(statistic=-17.431588891905882, pvalue=6.336089707623977e-42)


In [117]:
cust["pre_usage"].mean(), cust["Post_usage_1month"].mean(), cust["post_usage_2ndmonth"].mean(), cust["Latest_mon_usage"].mean()

(52.23, 52.775, 58.05250000000003, 63.17400000000001)

 Campaign has no immediate effect==> no effect after 1 monthe, but has effect after 2 months and in the latest month

## Q3
Is there any difference between males & females in terms of credit card usage? (
## NB:

In [118]:
# Ho: avg spending of Male= avg spending of Female (There's no d/ce)
# Ha: avg spending of Male!= avg spending of Female (There's d/ce)
# Test: t-test=> independence t-test (This is a test for the null hypothesis that 2 independent samples have identical average (expected) values.)
# Decision rule:
#                          Let CI=99% (i.e. alpha=0.01)
#                               if p-value> alpha, accept Ho
#                               if p-value< alpha, reject Ho and accept Ha


In [119]:
# ? stats.ttest_ind

In [120]:
cust.head(2)

custid  sex  AqChannel  region  Marital_status  segment  pre_usage  \
0      70    0          4       1               1        1         57   
1     121    1          4       2               1        3         68   

   Post_usage_1month  Latest_mon_usage  post_usage_2ndmonth  
0                 52              49.2                 57.2  
1                 59              63.6                 64.9

###  Before Campaign

In [121]:
s1_female=cust[cust["sex"]==1]
s2_male=cust[cust["sex"]==0]
stats.ttest_ind(s1_female["pre_usage"],s2_male["pre_usage"])

Ttest_indResult(statistic=-0.7480109580953392, pvalue=0.4553410655360075)

In [122]:
s1_female["pre_usage"].mean(),s2_male["pre_usage"].mean()

(51.73394495412844, 52.824175824175825)

### After 1 month of campaign

In [123]:
stats.ttest_ind(s1_female["Post_usage_1month"],s2_male["Post_usage_1month"])

Ttest_indResult(statistic=3.7340738531536797, pvalue=0.00024625461203549315)

In [124]:
s1_female["Post_usage_1month"].mean(),s2_male["Post_usage_1month"].mean()

(54.99082568807339, 50.120879120879124)

### After 2 months of campaign

In [125]:
stats.ttest_ind(s1_female["post_usage_2ndmonth"],s2_male["post_usage_2ndmonth"])

Ttest_indResult(statistic=3.7340738531536926, pvalue=0.0002462546120354815)

In [126]:
s1_female["post_usage_2ndmonth"].mean(),s2_male["post_usage_2ndmonth"].mean()

(60.48990825688068, 55.13296703296704)

### Latest month after campaign

In [127]:
stats.ttest_ind(s1_female["Latest_mon_usage"],s2_male["Latest_mon_usage"])

Ttest_indResult(statistic=-0.4129986492968787, pvalue=0.680054497423219)

In [128]:
s1_female["Latest_mon_usage"].mean(),s2_male["Latest_mon_usage"].mean()

(62.87339449541283, 63.53406593406595)

In [129]:
# conclusion ????

## Q4
Is there any difference between segments of customers in terms of credit card usage? 

In [130]:
cust["segment"].value_counts()

2    105
3     50
1     45
Name: segment, dtype: int64


## NB:
we can't use t-test as we have 3 differenet segment categories. Hence we go with ANOVA
* ANOVA test used to compare the means of more than 2 groups (t-test or Z-test can be used to compare 2 groups)
* Groups mean differences inferred by analyzing variances
* ANOVA uses variance-based F test to check the group mean equality.
### ANOVA Hypotheses
* Null hypothesis: Groups means are equal (no variation in means of groups)
H0: μ1=μ2=…=μp
* Alternative hypothesis: At least, one group mean is different from other groups
H1: All μ are not equal
* One-way/one factor ANOVA compares three or more levels (conditions) of one factor. 
* On the other hand, two-way ANOVA compares the effect of multiple levels of two factors. In one-way ANOVA, the number of observations need not be same in each group whereas it should be same in the case of two-way ANOVA.

In [131]:
cust.head(2)

custid  sex  AqChannel  region  Marital_status  segment  pre_usage  \
0      70    0          4       1               1        1         57   
1     121    1          4       2               1        3         68   

   Post_usage_1month  Latest_mon_usage  post_usage_2ndmonth  
0                 52              49.2                 57.2  
1                 59              63.6                 64.9

In [132]:
# Ho: avg(seg1)=avg(seg2)=avg(seg3)- no d/ce\
# Ha: avg(seg1)!=avg(seg2)!=avg(seg3)- atleast one has d/f avg 
# Test: ANOVA
# Decision rule:
#                       Let CI=99% (i.e. alpha=0.01)
#                            if p-value> alpha, accept Ho
#                            if p-value< alpha, reject Ho and accept Ha


In [133]:
seg1=cust[cust["segment"]==1]
seg2=cust[cust["segment"]==2]
seg3=cust[cust["segment"]==3]

In [134]:
stats.f_oneway(seg1["pre_usage"], seg2["pre_usage"], seg3["pre_usage"])

F_onewayResult(statistic=21.28234763340855, pvalue=4.283274139124812e-09)

In [135]:
seg1["pre_usage"].mean(), seg2["pre_usage"].mean(), seg3["pre_usage"].mean()

(49.75555555555555, 56.161904761904765, 46.2)

we need to accept Ha

## Q 5
Is there any relation between region & Segment?

In [136]:
cust["segment"].value_counts()

2    105
3     50
1     45
Name: segment, dtype: int64

In [137]:
cust["region"].value_counts()

2    95
3    58
1    47
Name: region, dtype: int64

In [138]:
# both are categ types
# Ho: there is no asso b/n seg and reg
# Ha: there is asso b/n seg and reg
# Test: chi-square test tests the null hypothesis that the categorical data has the given frequencies. 
 # Decision rule:
#                       Let CI=99% (i.e. alpha=0.01)
#                            if p-value> alpha, accept Ho
#                            if p-value< alpha, reject Ho and accept Ha


In [139]:
ch1= pd.crosstab(cust["segment"],cust["region"]) # Compute a simple cross-tabulation of two (or more) factors.

In [140]:
stats.chi2_contingency(ch1) # the first value is chi sq value,the second value in the ff output is the p-value

(16.604441649489342,
 0.0023066300908054713,
 4,
 array([[10.575, 21.375, 13.05 ],
        [24.675, 49.875, 30.45 ],
        [11.75 , 23.75 , 14.5  ]]))

p-val< alpha val ->there is asso b/n seg and reg 

### Q6
Is the relationship between card usage in the latest month and pre usage of campaign? 

In [141]:

### NB:
# Here the prob is to check the r/s b/n two cont values --> use correlation test
# Ho: there is no r/s b/n latest month and pre usage of campaign
# Ha: there is r/s b/n latest month and pre usage of campaign
# Test: corr
 # Decision rule:
#                       Let CI=99% (i.e. alpha=0.01)
#                            if p-value> alpha, accept Ho
#                            if p-value< alpha, reject Ho and accept Ha

In [142]:
cust[["Latest_mon_usage","pre_usage"]].corr()

Latest_mon_usage  pre_usage
Latest_mon_usage           1.00000    0.66228
pre_usage                  0.66228    1.00000

### or 


In [143]:
stats.pearsonr(cust["Latest_mon_usage"],cust["pre_usage"])

PearsonRResult(statistic=0.6622801251558604, pvalue=1.2767419295069158e-26)

### or

In [144]:
np.corrcoef(cust["Latest_mon_usage"],cust["pre_usage"])

array([[1.        , 0.66228013],
       [0.66228013, 1.        ]])

* In all cases, \
==> r= 0.66228 and p-val=1.2767419....e-26\
==> there is r/s

## The End